context/app/static/js/components/workspaces/WorkspaceLaunchStopButtons/WorkspaceLaunchStopButtons.tsx

```diff
@@ -24,12 +26,17 @@ function StopWorkspaceButton({
   handleStopWorkspace,
   button: ButtonComponent,
   isStoppingWorkspace,
+  checkMaxDatasets = false,
 }: Omit<WorkspaceButtonProps, 'disableLaunch' | 'disableStop'>) {
   const { toastError } = useSnackbarActions();
+
   const currentWorkspaceIsRunning = isRunningWorkspace(workspace);
-  if (!currentWorkspaceIsRunning) {
+  const currentWorkspaceHasMaxDatasets = checkMaxDatasets && isWorkspaceAtDatasetLimit(workspace);
+
+  if (!currentWorkspaceIsRunning || currentWorkspaceHasMaxDatasets) {
```

Comment:
A little confused here. We don't want to allow workspaces to stop the workspace if it has max datasets?

Reply:
This is some logic required for separating the use cases of `WorkspaceListItem`. This component is used both in the 'Workspaces' page, for selecting workspaces for deletion, and in the 'Add Datasets to Workspace' dialog, for selecting workspaces to add datasets to.

In the former case, a workspace should be selectable for deletion regardless of the # of attached datasets. In the latter, if a workspace is at the max # of datasets, it needs a new tooltip and the 'stop jobs' button should be removed (as it should not be selectable regardless of whether jobs are running).

Maybe this prop can be renamed to describe these use cases, instead of what's actually being checked, to make this clearer? (for example, something like `isInAddDatasetsDialog`?)

Reply:
Ah, I get it now. Can we just use the existing `disableStop` prop in `WorkspaceLaunchStopButtons`? Albeit `disableStop` is a misnomer as it actually prevents the button from being shown.

Reply:
If the true intended behavior is for the button to be visible, but disabled, I would rename the existing `disable*` props to `show*` and add a new disabled prop for general use.

Reply:
For this case, I'm hesitant to use a `disabled` prop because the `hasMaxDatasets` condition that requires disabling the button should also generate a specific tooltip. Using the `disabled` prop would mean checking for `hasMaxDatasets` in both components, unless the tooltip is also passed as a prop - is this preferable? Or is the `disabled` prop intended for future cases?

Reply:
I might try the following steps:

1. Add a `disabled` prop to `WorkspaceListItem`.
2. Add a `tooltip` prop to `WorkspaceListItem`. Do something like `const tooltipToDisplay = tooltip === undefined && isRunning ? 'Stop all jobs before selecting.' : tooltip` to prioritize tooltips.
3. Create a wrapper component for `WorkspaceListItem` in `AddDatasetsFromSearchDialog` to pass in the tooltip and disabled props.

Reply:
Nice!